src/index.js

```diff
@@ -0,0 +1,261 @@
+// import inquirer
+const inquirer = require("inquirer");
+const { writeToFile } = require("./util");
+const { questions } = require("./questions");
+
+//questions
+
+const getOtherContents = ({ installation, usage, tests }) => {
+  const contents = [];
+  if (installation) contents.push("- [Installation](#installation)");
+  if (usage) contents.push("- [Usage](#usage)");
+  if (tests) contents.push("- [Tests](#tests)");
+  return contents;
+};
+
+// constructing the dynamic README.md
+// project title
+const constructTitle = (projectTitle, license) => {
+  return `# ${projectTitle} ![${license}](https://img.shields.io/static/v1?label=${license}&message=Licence&color=<COLOR>)`;
+};
+
+// table of contents
+const generateTableOfContents = (installation, usage, tests) => {
+  const contents = [
+    "- [Description](#description)",
+    ...getOtherContents(installation, usage, tests),
+    "- [Contributing](#contributing)",
+    "- [License](#license)",
+    "- [Question](#question)",
+  ];
+  return `## Table of Contents\n
+${contents.join("\n")}
+`;
+};
+
+// project description
+const constructDesc = (projectDesc) => {
+  return `
+## Description
+
+${projectDesc}`;
+};
+
+// project installation
+const constructInstall = (installation) => {
+  if (installation) {
+    return `
+## Installation\n
+Follow these steps for installation:
+
+${installation
+  .map(function (installSteps) {
+    return `
+\`\`\`
+${installSteps.hasInstallScript}
+\`\`\``;
+  })
+  .join("\n")}`;
+  } else {
+    return "";
+  }
+};
+
+// project directions
+const constructDirections = (usage) => {
+  if (usage) {
+    return `
+## Usage
+
+${usage
+  .map(function (usageSteps) {
```

Comment:
Although this is correct, it would be more readable if this function was within a callback function because of the .join() afterwards.